### ISSUE TITLE
Dependency?

### ISSUE THREAD

**Issue by the872** [Rhernandez513/Comp-330-work#1]
@Rhernandez513 it was very witty of your to use an existing [project](https://github.com/edburnett/twitter-text-python) to finish the homework. However I would argue that you are not necessarily in control of your code. 

For the shorter-term prospect, I think you will be fine. But if we looked further in the future and for whatever reason Ed could not keep up with whatever updates Twitter may implement. For example, if Twitter changed their [character count limits](http://blogs.wsj.com/digits/2016/01/05/twitter-to-expand-tweets-140-character-limit-to-10000/), how would your project react to that? 

Let me be clear, there is nothing wrong with the project in respect to what was asked out of the homework but for the sake of habits, do you think it is better to maintain a greater control over your code?

**Comment by Rhernandez513** [comment]
[Unit tests built into original project](https://github.com/edburnett/twitter-text-python/blob/master/ttp/tests.py)

[Added some unit tests addressing possible cases mentioned in specification, as well as larger than 140 char count](https://github.com/Rhernandez513/Comp-330-work/commit/bccd3d512f25ffd99e353aa15c38977bba2900d1)

I recognize that the current design of the project is lacking concerning extensibility.  

I would disagree with your assertion that I am not in control of my code.  Considering it is an open source project, in the event that Ed could not, or would not merge a fix that addressed a stakeholder concern, it is simple enough to fork the project and edit/extend it to meet such a need.

What I'm trying to get across is that, DRY and KISS.  In a production system, I would use an approach closer to @ssmeke with proper generics and interfaces

**Comment by the872** [comment]
Fair and valid points. I think it comes down to personal preference and how you assess certain risks. [SOLID](https://en.wikipedia.org/wiki/SOLID_%28object-oriented_design%29) was was what came to mind when the dependency concern came about. 

What I did for my project was actually use the regex from Ed's project because it was well tested and thorough but I didn't feel at ease using the entire project as a dependency. But again, personal preference.

**Comment by csoulakian** [comment]
I agree with @the872, you didn't really make the code your own. I think the assignment was less about finding libraries that already have the functionality and more about going through the process of determining what functionalities need to be implemented and how to implement them. What's the point in even doing the project then? We could have just went to the twitter website to use their parser and call it a day.
